lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/MessagingTransportFactory.java

```diff
@@ -42,24 +41,12 @@
 
   private final String localAddress;
 
-  /**
-   * @deprecated Have an instance injected instead.
-   */
-  @Deprecated
   @Inject
   public MessagingTransportFactory(final LocalAddressProvider localAddressProvider) {
```

Comment:
I think so too, but we can't make it private right now, because one of constructors of `NameLookupClient` needs this constructor. I've tried multiple things to replace the `new` call with injection, and none of them worked, this PR is the best I could make work for now. I believe if we make `NameLookupClient` injectable we'll be able to make this constructor private indeed.